safe_video/ui/helper_classes.py

```diff
@@ -36,23 +57,34 @@ def upload_image(self, old_path: str, filename: str) -> str:
             str: returns the key where the image can be found
         """
         name, fmt = re.findall("(.*)\.(.*)$", filename)[0]
-        new_path = self.cache_path + name + "{}." + fmt
-        if not os.path.exists(self.cache_path):
+        if not os.path.exists(self.cache_path): # check if cache folder exists
             os.makedirs(self.cache_path)
         counter = 0
-        new_path = str(self.cache_path + name + "_{}." + fmt)
-        while os.path.isfile(new_path.format(counter)):
+        new_folder = str(self.cache_path + name + "_{}")
+        while os.path.isdir(new_folder.format(counter)):
             counter += 1
-        shutil.copy(old_path, new_path.format(counter))
-        path_name = name + '_' + str(counter)
-        self.__setitem__(path_name, Image(cache_path=self.cache_path, name=path_name, orig_name=name, format=fmt))
-        return path_name
+        id = f'{name}_{counter}' # unique id for this image
+        os.makedirs(new_folder.format(counter))
+        new_path = f'{new_folder.format(counter)}/{self.ORIGINAL_NAME}.{fmt}'
+        shutil.copy(old_path, new_path)
+        img = PIL.Image.open(new_path)
+        img = img.resize((100, 100))
```

Comment:
This will stretch images that are not rectengular which we probably dont want.
If we use a larger size we can also use this as the image we show which could improve loading times for large images